src/System/Random/Stateful.hs

```diff
@@ -642,6 +641,46 @@ runSTGen_ g action = fst $ runSTGen g action
 -- >   thawGen = MWC.restore
 -- >   freezeGen = MWC.save
 --
+-- === @FrozenGen@
+--
+-- `FrozenGen` gives us ability to use any stateful pseudo-random number generator in its
+-- immutable form, if one exists that is. This concept is commonly known as a seed, which
+-- allows us to save and restore the actual mutable state of a pseudo-random number
+-- generator. The biggest benefit that can be drawn from a polymorphic access to a
+-- stateful pseudo-random number generator in a frozen form is the ability to serialize,
+-- deserialize and possibly even use the stateful generator in a pure setting without
+-- knowing the actual type of a generator ahead of time. For example we can write a
+-- function that accepts a frozen state of some pseudo-random number generator and
+-- produces a short list with random even integers.
+--
+-- >>> import Data.Int (Int8)
+-- >>> :{
+-- myCustomRandomList :: FrozenGen f m => f -> m [Int8]
```

Comment:
I think there was a PR just yesterday to try to get rid of uncommon types like `Int8`. Would this example work just as well with `Int`? If so, I vote to use `Int` instead.

Reply:
I totally disagree that `Int8` is uncommon. It is not exported by default from `Prelude` and that is very unfortunate, but that is no reason to avoid it.

Reason why I used `Int8`, because I'd rather see a short list like that in haddock: `[-50,-2,4,-8,-58,-40,24,-32,-110,24]` then some ugly thing like that: `[-2069219337697363530,2082786383697626148,-4827225046748036658,7360998562723210750,-8084130566829248252,2997296016991233784,1279194016157494726,-6102362765004597032,-4998293769779175912,-6089568709720028704]`

A few places where the switch was made from `Int8` -> `Int` I do agree with. When you do `uniformRM (1, 6)`, there is no point to use `Int8`, because you are already restricting the range and in Haskell both `Int8` and `Int` underneath have the same size (at least for now)